### PR TITLE
WindowServer: Menu keyboard interaction improvements

### PR DESCRIPTION
**Pull request by codebutler**

### CODE REVIEW

**Comment by awesomekling** [comment]
Great ideas! Please make sure commit messages are wrapped at 72 characters :)

**Comment by codebutler** [comment]
> Great ideas! Please make sure commit messages are wrapped at 72 characters :)

Whoops! Fixed that and another accidental change :)

File: Userland/Services/WindowServer/Menu.cpp [hunk 1]
```diff
@@ -370,15 +370,26 @@ void Menu::event(Core::Event& event)
...
         if (!hovered_item()) {
-            int counter = 0;
-            for (const auto& item : m_items) {
-                if (item.type() != MenuItem::Separator && item.is_enabled()) {
-                    set_hovered_index(counter, key == Key_Right);
-                    break;
+            if (key == Key_Up) {
+                // Default to the last enabled, non-separator item on key press if one has not been selected yet
+                for (auto i = static_cast<int>(m_items.size()) - 1; i >= 0; i--) {
+                    auto& item = m_items.at(i);
```
Comment (codebutler) [inline Userland/Services/WindowServer/Menu.cpp:377]: Is there a more idiomatic way to iterate in reverse?

Comment (awesomekling) [reply Userland/Services/WindowServer/Menu.cpp:377]: Not yet. It would be nice to have something for that :)

**Pull request merged**